### PR TITLE
Add entrance and exit criteria to release template

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Add entrance and exit criteria to release template

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Hi @bbarani @CEHENKLE @dblock Please review this PR to update the release issue template with entrance and exit criteria. 
Wondering if entrance criteria should be placed in component issue rather than main issue.
Thanks!

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 1]
```diff
@@ -20,6 +20,32 @@ This issue captures the state of the OpenSearch release, its assignee (Release M
...
+-- | -- | -- | --
+Documentation draft PRs are up and in tech review for all component changes | :red_circle: |   |
+Sanity testing is done for all components | :red_circle: |   |
+Code coverage has not decreased (all new code has tests) | :red_circle: |   |
+Release notes are ready and available for all components | :red_circle: |   |
+Roadmap is up-to-date (information is available to create release highlights) | :red_circle: |   |
+Release ticket is cut, and there's a forum post announcing the start of the window | :red_circle: |   |
+[Any necessary security reviews](##Security-Reviews) are complete | :red_circle: |   |
+
+### OpenSearch {{ env.VERSION }} [exit criteria](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#exit-criteria-to-close-release-window) status:
```
Comment (bbarani) [inline .github/ISSUE_TEMPLATE/release_template.md:34]: @CEHENKLE Should we add "No unpatched vulnerabilities of medium or higher severity that have been publicly known for more than 60 days" to exit criteria as well?

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 2]
```diff
@@ -20,6 +20,32 @@ This issue captures the state of the OpenSearch release, its assignee (Release M
...
+All integration tests are passing |  :red_circle: |   |
+Release blog is ready | :red_circle: |   |
+
+### OpenSearch-Dashboards {{ env.VERSION }} [exit criteria](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#exit-criteria-to-close-release-window) status:
+Criteria | Status | Description  | Comments
+-- | -- | -- | --
+Documentation has been fully reviewed and   signed off by the documentation community | :red_circle: |   |
+All integration tests are passing | :red_circle: |   |
+Release blog is ready | :red_circle: |   |
+
```
Comment (CEHENKLE) [inline .github/ISSUE_TEMPLATE/release_template.md:48]: Can we add something about all med and high CVEs older than 60 days have been addressed?

Comment (CEHENKLE) [reply .github/ISSUE_TEMPLATE/release_template.md:48]: I think we need to add that to the releasing.md as well.

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 1 above)
Comment (CEHENKLE) [reply .github/ISSUE_TEMPLATE/release_template.md:34]: I put it into exit criteria, but yes :)

**Review by CEHENKLE — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 2 above)
Comment (gaiksaya) [reply .github/ISSUE_TEMPLATE/release_template.md:48]: Should this be in main issue as well as child issues? (Basically build repo issue and all components issues as well?)

Comment (gaiksaya) [reply .github/ISSUE_TEMPLATE/release_template.md:49]: Added to the main issue for now.

Comment (CEHENKLE) [reply .github/ISSUE_TEMPLATE/release_template.md:50]: I think it makes sense to push it out to all the repos, but WDYT?

**Review by CEHENKLE — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 2 above)
Comment (gaiksaya) [reply .github/ISSUE_TEMPLATE/release_template.md:50]: I believe @bbarani raised a point that if some criteria changes, how do we keep track of all places to change it. Its like how we update dates. We mention them in main issues and refer the child components to go look there. 
If the criteria is not gonna be that dynamic then I am okay to add them in child issues too for now.

**Review by bbarani — APPROVED** [review]

**Pull request merged**